src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs

```diff
@@ -3541,6 +3546,13 @@ void BuildCustomMarkupExtensionPropertyValue(IIndentedStringBuilder writer, Xaml
 			}
 		}
 
+		private bool IsNotFrameworkElementButNeedsSourceLocation(XamlObjectDefinition objectDefinition)
+		{
+			return objectDefinition.Type.Name == "VisualState"
+				|| objectDefinition.Type.Name == "AdaptiveTrigger"
+				|| objectDefinition.Type.Name == "StateTrigger";
+		}
```

Comment:
nit: Make static and reduce repeating `objectDefinition.Type.Name`

```suggestion
		private static bool IsNotFrameworkElementButNeedsSourceLocation(XamlObjectDefinition objectDefinition)
			=> objectDefinition.Type.Name is "VisualState" or "AdaptiveTrigger" or "StateTrigger";
```